backend/core/detection.py

```diff
@@ -830,6 +844,33 @@ def mark_outdated(self, hij_key: str, redis_hij_key: str) -> NoReturn:
                 msg, exchange=self.hijack_exchange, routing_key="outdate", priority=1
             )
 
+        def gen_implicit_withdrawal(self, monitor_event: Dict) -> NoReturn:
+            """
+            Checks if a benign BGP update should trigger an implicit withdrawal
+            """
+            # log.debug('{}'.format(monitor_event['key']))
+            prefix = monitor_event["prefix"]
+            peer_asn = monitor_event["peer_asn"]
+            if self.redis.exists("prefix_{}_peer_{}_hijacks".format(prefix, peer_asn)):
+                # generate implicit withdrawal
+                withdraw_msg = {
+                    "service": "implicit-withdrawal",
```

Comment:
I really really don't like this hacky approach of sending it as a monitor :/

Reply:
It is not really a hack, but code reuse :) ; we need to somehow trigger implicit withdrawals when the paths change. I do not see any other good way of doing this without changing large chunks of code (the current solution takes advantage of what ARTEMIS does well, which is processing updates quickly, without requiring changes in the logic, besides the pseudo-withdrawal messages). However, any other ideas on how to achieve the same end result are welcome ;) .